homework_1/task3/task3.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/sethgrid/pester"
+)
+
+type pokemonAPIResponse struct {
+	Name                     string
+	Location_area_encounters string
```

Comment:
Nice workaround with the naming so JSON can be correctly marshalled. I believe you'll agree with me that it's not the cleanest way to do it since snake-case isn't a great convention to name variables with (at least not in Go).

One thing we didn't go through in the lecture are struct tags which allow you to add some metadata to your structs. Via struct tags you can add some metadata which tells the `json` package what attribute name in the JSON to map to this attribute in Go struct.

If you take a look at the docs of `json.Marshal` I believe you'll figure out how to use the tags to make this cleaner! https://golang.org/pkg/encoding/json/#Marshal